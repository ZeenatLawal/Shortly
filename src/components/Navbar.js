import React from 'react';
import Navbar from 'react-bootstrap/Navbar';
import Nav from 'react-bootstrap/Nav';
import Container from 'react-bootstrap/Container';
import '../styles/Navbar.css';
import logo from '../images/logo.svg';

const Navpanel = () => (
  <Navbar collapseOnSelect expand="lg" bg="light" variant="light">
    <Container>
      <Navbar.Brand><img src={logo} alt="Shortly Logo" /></Navbar.Brand>
      <Navbar.Toggle aria-controls="responsive-navbar-nav" />
      <Navbar.Collapse id="responsive-navbar-nav">
        <Nav className="align-items-center">
          <Nav.Link>Features</Nav.Link>
          <Nav.Link>Pricing</Nav.Link>
          <Nav.Link>Resources</Nav.Link>
        </Nav>
        <Nav className="d-lg-none">
          <hr />
        </Nav>
        <Nav className="align-items-center">
          <Nav.Link>Login</Nav.Link>
          <Nav.Link>Sign Up</Nav.Link>
        </Nav>
      </Navbar.Collapse>
    </Container>
  </Navbar>
);

export default Navpanel;
