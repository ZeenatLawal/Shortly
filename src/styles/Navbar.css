.navbar-light .navbar-toggler {
  color: white;
  border-color: white;
}

.navbar {
  position: fixed;
  width: 100%;
  top: 0;
  background-color: white;
  padding-bottom: 0;
}

.navbar-collapse {
  background-color: var(--dark-violet);
  border-radius: 4%;
  padding: 1rem 0;
}

.navbar-light .navbar-nav .nav-link {
  color: white;
}

hr {
  color: var(--gray);
  width: 90%;
  margin: 1rem auto;
}

.sign-up {
  padding: 0 5rem;
}

@media screen and (min-width: 1024px) {
  .navbar {
    padding-top: 0;
  }

  .navbar-collapse {
    background-color: white;
    justify-content: space-between;
  }

  .navbar-light .navbar-nav .nav-link {
    color: var(--grayish-violet);
    padding: 0.5rem 1rem;
  }

  .navbar-light .navbar-nav .nav-link:hover {
    color: var(--darker-violet);
  }

  .sign-up {
    padding: 0 1rem;
    margin-left: 1rem;
  }
}
