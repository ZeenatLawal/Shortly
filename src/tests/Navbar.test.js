import renderer from 'react-test-renderer';
import Navbar from '../components/Navbar';

describe('Navbar component', () => {
  test('Snapshot test', () => {
    const navbar = renderer.create(
      <Navbar />,
    )
      .toJSON();
    expect(navbar).toMatchSnapshot();
  });
});
