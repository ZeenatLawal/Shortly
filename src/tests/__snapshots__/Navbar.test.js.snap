// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar component Snapshot test 1`] = `
<nav
  className="navbar navbar-expand-lg navbar-light"
>
  <div
    className="container"
  >
    <span
      className="navbar-brand"
    >
      <img
        alt="Shortly Logo"
        src="logo.svg"
      />
    </span>
    <button
      aria-controls="responsive-navbar-nav"
      aria-label="Toggle navigation"
      className="navbar-toggler collapsed"
      onClick={[Function]}
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <div
      aria-expanded={null}
      className="navbar-collapse collapse"
      id="responsive-navbar-nav"
    >
      <div
        className="align-items-center navbar-nav"
        onKeyDown={[Function]}
      >
        <a
          className="nav-link"
          data-rr-ui-event-key={null}
          href="#"
          onClick={[Function]}
          onKeyDown={[Function]}
          role="button"
          tabIndex={0}
        >
          Features
        </a>
        <a
          className="nav-link"
          data-rr-ui-event-key={null}
          href="#"
          onClick={[Function]}
          onKeyDown={[Function]}
          role="button"
          tabIndex={0}
        >
          Pricing
        </a>
        <a
          className="nav-link"
          data-rr-ui-event-key={null}
          href="#"
          onClick={[Function]}
          onKeyDown={[Function]}
          role="button"
          tabIndex={0}
        >
          Resources
        </a>
      </div>
      <div
        className="d-lg-none navbar-nav"
        onKeyDown={[Function]}
      >
        <hr />
      </div>
      <div
        className="align-items-center navbar-nav"
        onKeyDown={[Function]}
      >
        <a
          className="nav-link"
          data-rr-ui-event-key={null}
          href="#"
          onClick={[Function]}
          onKeyDown={[Function]}
          role="button"
          tabIndex={0}
        >
          Login
        </a>
        <div
          className="sign-up"
        >
          <a
            className="text-white nav-link"
            data-rr-ui-event-key={null}
            href="#"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={0}
          >
            Sign Up
          </a>
        </div>
      </div>
    </div>
  </div>
</nav>
`;
